### ISSUE TITLE
ShouldNotHaveValidationErrorFor should output validation errors

### ISSUE THREAD

**Issue by jzabroski** [JeremySkinner/FluentValidation#909]
This is the stack trace I get in Resharper Unit Test Session:
```
FluentValidation.TestHelper.ValidationTestException : Expected no validation errors for property Name
   at FluentValidation.TestHelper.ValidationResultTester`2.ShouldNotHaveValidationError(IEnumerable`1 properties) in C:\Projects\FluentValidation\src\FluentValidation\TestHelper\ValidationResultTester.cs:line 73
```
It should say:
```
Expected no validation errors for property Name
----
Validation Errors:
  [0]: error message 1
  [n]: error message n
```
This helps figure out which rules were triggered.

**Comment by jzabroski** [comment]
https://github.com/JeremySkinner/FluentValidation/blob/253faf602dfd6103dcb0773c40392c60969415c3/src/FluentValidation/TestHelper/ValidationResultTester.cs#L61-L62

should be:
```csharp
    if (!failures.Any())
    {
        var errorMessageBanner = $"Expected a validation error for property {propertyName}\n";
        string errorMessageDetails;
        var i = 0;
        foreach (var failure in failures)
        {
            errorMessageDetails += $"\t[{i}]: {failure.ErrorMessage}\n";
        }
 	throw new ValidationTestException(errorMessageDetails);
    }
```

**Comment by JeremySkinner** [comment]
Thanks for the suggestion - It’s a good idea. I’ll put it on the backlog, although it’s probably going to be a while until I can get to this.

**Comment by jzabroski** [comment]
Can I submit the above as a patch, or is there more work to do?

**Comment by JeremySkinner** [comment]
If you’d like to submit it as a pull request then I’d be happy to review it. I expect It’ll also require several tests to be changed too. Thanks.

**Comment by JeremySkinner** [comment]
Added by PR in 472e55f.

**Comment by jzabroski** [comment]
Thanks, Jeremy! Hope all is well with you.
